pkg/test/framework/components/istio/operator.go

```diff
@@ -400,11 +400,22 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 			// before the external control plane cluster. Since remote clusters use gateway injection, we can't install the gateways
 			// until after the control plane is running, so we install them here. This is not really necessary for pure (non-config)
 			// remote clusters, but it's cleaner to just install gateways as a separate step for all remote clusters.
-			if err = installRemoteClusterGateways(s, i, c); err != nil {
-				return i, err
+
+			// Config cluster may have a separate config from pure remote clusters.
+			configFiles := []string{
+				filepath.Join(testenv.IstioSrc, IntegrationTestRemoteGatewaysIOP),
+			}
+			if c.IsConfig() {
+				configFiles = append(configFiles, istioctlConfigFiles.remoteIopFile)
```

Comment:
This seems strange. Why is it using the `remoteIopFile` (not `configIopFile`) for config clusters, and doesn't need anything for pure remote clusters?

Reply:
can't use `configIopFile` because we are explicitly disabling the gateways there and it is a requirement to setup config cluster. We don't need anything for pure remote clusters cuz we already have a dedicated yaml for that (https://github.com/istio/istio/blob/master/tests/integration/iop-remote-integration-test-gateways.yaml) 
@stevenctl  This piece of code is only running for remote clusters so I don't think there's a need for `IsRemote`.

Reply:
`remoteIopFile` is needed because it has the info about `externalCA` with enabled gateways. IIUC Here we are not setting pure config cluster, we are setting remote config cluster (i.e. not the config part of the config cluster)

Reply:
If I can attempt to summarize why we're changing things here:

1. We install on all remotes (config + pure remote)
2. We later install gateway on remote clusters (this must be done in a separate step)
3. That gateway install part gets rid of the external CA settings

This PR:
4. Overlays the external CA stuff when installing gateways. 

Instead we might:
1. Expose some way to set remote cluster gateway overlays the same way we expos ControlPlaneValues, RemoteClusterValues etc.

Reply:
I agree!! we should have a more generic way to do this.

Reply:
Yeah, this is kind of subtle ... I missed the part about `remoteIopFile` using `controlPlaneValues`. I agree that a good solution would be to enable setting gateway overlays in the test config.

Otherwise, I still wonder why you need to use `remoteIopFile` instead of `configIopFile` given that you are also overlaying the external CA stuff there too: `cfg.ConfigClusterValues = cfgConfigYaml`. Seems like using `remoteIopFile` when installing the gateways could be removing something that should be on the config cluster??

Reply:
We need to remove the gateway installation initially when it tries to setup the config cluster, we are using this config in `configIopFile`. And when it comes to gateway installation in later stages, it still has `enabled: false` for that.
https://github.com/istio/istio/pull/39073/files#diff-dad3a0fd21c90716b69a568d6e74151665c33ec7492869d8c2b46edc57eb2e45R129

Reply:
> in later stages, it still has enabled: false 

Yes, but the gateway yaml changes it to true: https://github.com/istio/istio/blob/dde2015562d8bf0216237322a15dd4625b3530c2/tests/integration/iop-remote-integration-test-gateways.yaml#L11

Is the problem that the override goes in the wrong direction ... `configIopFile` overrides the gateway one, instead the other way around?

I also just noticed that your override for remote clusters is enabling pilot: https://github.com/istio/istio/blob/98336b09e94298a4a9c3f8fa9126023c1b865023/tests/integration/security/external_ca/main_test.go#L136

So that will install istiod now on non-config remote clusters. Normally the remote clusters would also be istiodless.

Reply:
yeah, I guess you're right. Let me try to use  `iop-remote-integration-test-gateways.yaml` on top of `configIopFile`